invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -0,0 +1,125 @@
+
+    funderName = fields.Str(attribute="funder.name")
+    funderIdentifier = fields.Str(attribute="funder.identifier")
+    funderIdentifierType = fields.Str(attribute="funder.scheme")
+    awardTitle = fields.Str(attribute="award.title")
+    awardNumber = fields.Str(attribute="award.number")
+    # PIDS-FIXME: URI should be processed depending on the schema
+    awardURI = fields.Str(attribute="award.identifier")
+
+    @post_dump(pass_many=False)
+    def uppercase(self, data, **kwargs):
+        """Upper case the type."""
+        if data.get("funderIdentifierType"):
+            upper_type = data["funderIdentifierType"].upper()
+            data["funderIdentifierType"] = upper_type
+
+        return data
+
+
+class CSLJSONSchema(Schema):
+    """DataCite 4.3 Marshmallow Schema."""
+
+    publisher = fields.Str(attribute='metadata.publisher')
+    DOI = fields.Method('get_DOI')
+    language = fields.Method('get_language')
+    title = fields.Method('get_title')
+    issued = fields.Method('get_issued')
+    abstract = fields.Method('get_abstract')
+    author = fields.List(
+        fields.Nested(CreatorSchemaCSL), attribute='metadata.creators')
+    note = fields.List(
+        fields.Nested(FundingSchemaCSL), attribute='metadata.funding')
+    version = SanitizedUnicode(attribute="metadata.version")
+    # PIDS-FIXME: What about versioning links and related ids
+    type = fields.Method('get_type')
+    id = fields.Str(attribute='metadata.id')
+
+    def get_DOI(self, obj):
+        """Get DOI."""
+        metadata = obj["metadata"]
+        identifiers = metadata.get("identifiers", [])
+        for id_ in identifiers:
+            if id_["scheme"] == "DOI":
+                return id_["identifier"]
+
+    def get_language(self, obj):
+        """Get language."""
+        metadata = obj["metadata"]
+        languages = metadata.get("languages")
+
+        if languages:
+            # PIDS-FIXME: How to choose? the first?
+            return languages[0]["id"]
+
+        return missing
+
+    def get_title(self, obj):
+        """Get title."""
+        metadata = obj["metadata"]
+        return metadata.get("title", missing)
+
+    def get_issued(self, obj):
+        """Get issued dates."""
+        date_parts = []
+        publication_date = obj["metadata"].get("publication_date", "")
+        for splitedDate in publication_date.split("/"):
+            date_parts.append(splitedDate.split("-"))
+
+        return {"date-parts": date_parts}
+
+    def get_abstract(self, obj):
+        """Get abstract."""
+        metadata = obj["metadata"]
+        return metadata.get("description", missing)
+
+    def get_type(self, obj):
+        """Get resource type."""
+        resource_type = obj["metadata"]["resource_type"]
+        resource_type_record = self._read_resource_type(resource_type["id"])
+        props = resource_type_record["props"]
+        return props.get("datacite_general", "Other")
+
+
+    def _read_resource_type(self, id_):
```

Comment:
I'm not convinced about doing this call here, but it does not break the layers so is good. However, it is also used in datacite serializer, you might want to move the function to the main `serializers` folder (e.g. in the `__init__` or in a `serializers.py` file, please NO `utils.py`/`helpers.py` hahaha)